File: Algorithms/Implementation/Lisa's Workbook.php
<?php

/*
 * Complete the 'workbook' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER n
 *  2. INTEGER k
 *  3. INTEGER_ARRAY arr
 */

function workbook($n, $k, $arr) {
    # Write your code here
    $count = 0;
    $page = 1;

    for ($i = 0; $i < $n; $i++) {
        $problems = $arr[$i];

        for ($j = 1; $j <= $problems; $j++) {
            if ($j == $page) {
                $count++;
            }
            if ($j % $k == 0 || $j == $problems) {
                $page++;
            }
        }
    }
    return $count;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$first_multiple_input = explode(' ', rtrim(fgets(STDIN)));
$n = intval($first_multiple_input[0]);
$k = intval($first_multiple_input[1]);

$arr_temp = rtrim(fgets(STDIN));
$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));
$result = workbook($n, $k, $arr);

fwrite($fptr, $result . "\n");
fclose($fptr);

File: Algorithms/Implementation/Lisa's Workbook.py
import os

#
# Complete the 'workbook' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER n
#  2. INTEGER k
#  3. INTEGER_ARRAY arr
#

def workbook(n, k, arr):
    # Write your code here
    count = 0
    page = 1

    for problems in arr:
        for current_problem in range(1, problems + 1):
            if current_problem == page:
                count += 1
            if current_problem == problems or current_problem % k == 0:
                page += 1

    return count

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    first_multiple_input = input().rstrip().split()
    n = int(first_multiple_input[0])
    k = int(first_multiple_input[1])

    arr = list(map(int, input().rstrip().split()))
    result = workbook(n, k, arr)

    fptr.write(str(result) + '\n')
    fptr.close()
